lib/imsg.c

```diff
@@ -310,6 +310,7 @@ imsg_get_fd(struct imsgbuf *ibuf)
 	fd = ifd->fd;
 	TAILQ_REMOVE(&ibuf->fds, ifd, entry);
 	free(ifd);
+	TAILQ_FIRST(&ibuf->fds) = NULL;
```

Comment:
Same comment here.  We need to have a for loop that saves the next pointer in imsg_clear